sklearn/linear_model/ridge.py

```diff
@@ -327,7 +327,7 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
         y = check_array(y, dtype=np.float64, ensure_2d=False, order='F')
     else:
         X = check_array(X, accept_sparse=['csr', 'csc', 'coo'],
-                        dtype=np.float64)
+                        dtype='numeric')
```

Comment:
I went with `numeric` because I thought it was more generic. I think you are right though and we should go for `[np.float64, np.float32]` instead. However, the dtype consistency test should fail if you stick with `np.float64`.